optuna/storages/_redis.py

```diff
@@ -73,6 +73,15 @@ def __init__(self, url: str) -> None:
         self._url = url
         self._redis = redis.Redis.from_url(url)
 
+    def __getstate__(self) -> Dict[Any, Any]:
+        state = self.__dict__.copy()
+        del state["_redis"]
+        return state
+
+    def __setstate__(self, state: Dict[Any, Any]) -> None:
+        self.__dict__.update(state)
+        self._redis = redis.Redis.from_url(self._url)
```

Comment:
This was to support `scheduler="processes"` with the Redis storage but I'm feeling that this multiprocessing scheduling requires more time and tests, at least for Redis. We can separate this PR into two, to the scheduler and multiprocess in memory storage. We could also leave the Redis storage outside the scope.